extensions/default/src/Toolbar/ToolbarLayoutSelector.tsx

```diff
@@ -79,7 +79,15 @@ function LayoutSelector({
         ],
       });
     }
-    viewportGridService.setLayout({ numRows, numCols });
+
+    // When a new layout is selected, keep any extra/offscreen viewports
+    // so that if any of those viewports were populated via the UI then they
+    // will be maintained in case those viewports are redisplayed later.
+    viewportGridService.setLayout({
+      numRows,
+      numCols,
+      keepExtraViewports: true,
```

Comment:
Lets just assume the extra viewport information is always stored on any setLayout change, then the API doesn't have to be updated, which reduces the number of changes here, and it doesn't hurt to store it.

Reply:
@wayfarer3130 we cannot assume that because it would break MPR and TMTV hanging protocols.